crates/ruff/tests/direxist_guard.rs

```diff
@@ -0,0 +1,25 @@
+#![cfg(not(target_family = "wasm"))]
+
+use std::env::set_current_dir;
+use std::process::Command;
+
+use insta_cmd::{assert_cmd_snapshot, get_cargo_bin};
+const BIN_NAME: &str = "ruff";
+
+#[test]
+fn check_in_deleted_directory_errors() {
+    let temp_dir = tempfile::tempdir().unwrap();
+    let temp_path = temp_dir.path().to_path_buf();
+    set_current_dir(&temp_path).unwrap();
+    drop(temp_dir);
+
+    assert_cmd_snapshot!(Command::new(get_cargo_bin(BIN_NAME)).arg("check"), @r###"
```

Comment:
I agree with @ntBre that changing the working directory is problematic. Let's only override the current working directory for the spanned CLI command.

```suggestion
    assert_cmd_snapshot!(Command::new(get_cargo_bin(BIN_NAME)).arg("check").current_dir(&temp_path), @r###"
```

Reply:
I think I tried `current_dir` when I looked at this, and it caused a different panic in the `Command` invocation, separate from the one we're trying to test. But it would be good to double check, and maybe I missed something.

Reply:
Yes, the `assert_cmd_snapshot` or `current_dir` calls an `unwrap()` somewhere. The test fails with: 

```bash
[...] called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory"
```

Reply:
Hmm I see. Let's add some module level documentation (or at least test level) explaining why this test *has to be its own module*.